addons/gamification/models/gamification_karma_tracking.py

```diff
@@ -5,65 +5,90 @@
+        End date is set to last day of current month, a millisecond before next month. """
+        from_date = date_utils.start_of(from_date, "month").date()
+        to_date = date_utils.end_of(from_date, "month") + relativedelta(days=1, microseconds=-1)
         select_query = """
-SELECT user_id,
-(
-    SELECT old_value from gamification_karma_tracking old_tracking
-    WHERE old_tracking.user_id = gamification_karma_tracking.user_id
-        AND tracking_date::timestamp BETWEEN %(from_date)s AND %(to_date)s
-        AND consolidated IS NOT TRUE
-        ORDER BY tracking_date ASC LIMIT 1
-), (
-    SELECT new_value from gamification_karma_tracking new_tracking
-    WHERE new_tracking.user_id = gamification_karma_tracking.user_id
-        AND tracking_date::timestamp BETWEEN %(from_date)s AND %(to_date)s
-        AND consolidated IS NOT TRUE
-        ORDER BY tracking_date DESC LIMIT 1
-)
-FROM gamification_karma_tracking
-WHERE tracking_date::timestamp BETWEEN %(from_date)s AND %(to_date)s
-AND consolidated IS NOT TRUE
-GROUP BY user_id """
+  SELECT DISTINCT user_id,
+		  FIRST_VALUE(old_value) OVER W_PARTITION as old_value,
+		  LAST_VALUE(new_value) OVER W_PARTITION as new_value,
+		  FIRST_VALUE(tracking_date) OVER W_PARTITION as from_tracking_date,
+		  LAST_VALUE(tracking_date) OVER W_PARTITION as to_tracking_date
+    FROM gamification_karma_tracking
+   WHERE tracking_date BETWEEN %(from_date)s AND %(to_date)s AND consolidated IS NOT TRUE
+  WINDOW W_PARTITION AS (PARTITION BY user_id ORDER BY tracking_date, id RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING)"""
         self.env.cr.execute(select_query, {
             'from_date': from_date,
-            'to_date': end_date,
+            'to_date': to_date,
         })
         results = self.env.cr.dictfetchall()
         if results:
             for result in results:
+                from_tracking_date = result.pop('from_tracking_date')
+                to_tracking_date = result.pop('to_tracking_date')
                 result['consolidated'] = True
-                result['tracking_date'] = fields.Date.to_string(from_date)
+                # Using latest tracking date prevents having tracking_date duplicates in case this method
+                # is called several times on the current ongoing month (although this is not the intended use
+                # of this method).
+                result['tracking_date'] = to_tracking_date
+                result['reason'] = _('Consolidation from %s to %s') % (from_tracking_date.strftime("%Y-%m-%m %H:%M:%S"), to_tracking_date.strftime("%Y-%m-%m %H:%M:%S"))
             self.create(results)
 
             self.search([
                 ('tracking_date', '>=', from_date),
-                ('tracking_date', '<=', end_date),
+                ('tracking_date', '<=', to_date),
                 ('consolidated', '!=', True)]
             ).unlink()
         return True
+
+    def _compute_originated_by(self):
```

Comment:
Hello David,
I will move the compute, sure!
The originated_by is used as the origin may differ and the link m2O is made by an inherit in website_slides and website_forum. This field is used in order to only display one single column in views (instead of having an OXO view with the two inherits)